modules/lm-coursier/src/main/scala/lmcoursier/CoursierConfiguration.scala

```diff
@@ -111,5 +172,53 @@ object CoursierConfiguration {
       credentials,
       Option(logger),
       Option(cache),
+      Option(ivyHome),
+      Option(followHttpToHttpsRedirections),
+      Option(strict),
+      extraProjects,
+      forceVersions,
+      reconciliation,
+      classpathOrder,
+      verbosityLevel,
+      Some(ttl),
+      checksums,
+      cachePolicies,
     )
+
+  private[lmcoursier] def cachePolicyToString(policy: CachePolicy): String = {
+    val sn: String = policy.getClass.getSimpleName
+    if (sn.endsWith('$' :: Nil)) {
+      sn.substring(0, sn.length-1)
+    } else {
+      sn
+    }
+  }
+
+  private val availableCachePolicies = Seq(
+    /* We need to reference these here, in order to avoid "guessing" or playing games with the classloader before
+    * or after shading, etc.
+    *
+    * As a side effect, it is possible to omit a CachePolicy from the public, if necessary. Conversely, a new
+    * CachePolicy should be exposed explicitly here.
+    */
+    CachePolicy.LocalOnly,
+    CachePolicy.LocalOnlyIfValid,
+    CachePolicy.LocalUpdateChanging,
+    CachePolicy.LocalUpdate,
+    CachePolicy.UpdateChanging,
+    CachePolicy.Update,
+    CachePolicy.FetchMissing,
+    CachePolicy.ForceDownload
+  )
+    .map(cp => (cachePolicyToString(cp), cp))
+    .toMap
+
+  private[lmcoursier] def cachePolicyFromString(name: String): CachePolicy = {
```

Comment:
Let's use [the method from coursier](https://github.com/coursier/coursier/blob/904670c8ab47c7fbc4c9c28aeeb4a63a71ce80d9/modules/cache/jvm/src/main/scala/coursier/parse/CachePolicyParser.scala#L9) for that, if going from `String`s to `CachePolicy` is really necessary.

Reply:
Won't be thanks to the ADT indeed.